python/pyspark/sql/udf.py

```diff
@@ -33,6 +33,10 @@ def _wrap_function(sc, func, returnType):
 
 
 def _create_udf(f, returnType, evalType):
+    from pyspark.sql.utils import _require_minimum_pyarrow_version
+
+    _require_minimum_pyarrow_version()
```

Comment:
We can't put this here because `_create_udf` is used for `udf` (not `pandas_udf`) as well. It is not related to pyarrow and we should be able to define normal udf without pyarrow.

Reply:
Yeah, that is not good!  I was a little hesitant to put it in `def pandas_udf` because things are a little different when used as a decorator.  How about leave it in `_create_udf` only when the eval type is a Pandas form?

Reply:
Yeah, sounds good.